nrc_ws/src/nrc_nav/src/nrc_drive_dr.py

```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env python
+
+import rospy
+
+import time
+import csv
+from numpy import genfromtxt
+
+from nrc_msgs.msg import DriveCommand
+
+# Global robot object
+command_pub = None
+# global time
+start_time = time.time()
+# time index for trajectory gen instructions
+instruction_index = 0
+# instructions pulled in at the start of code running
+instructions = None
+
+
+def generate_drive_command(timer_event):
+    # make sure we can access the global time
+    global start_time
+    # get local time at every tick to compare to the path-gen (in seconds)
+    local_time = time.ctime(start_time)
+
+    # use local time to update most recent instruction
+    while instructions[instruction_index + 1][0] < local_time:
+        # if current time has passed the next instruction's time, switch "most recent" to it
+        instruction_index += 1
+
+    # pull variables we need from most recent instruction
+    instruction_time = my_data[instruction_index][0]
+    # can get x and y from positions 1 and 2 when needed
+    vel = my_data[instruction_index][3]
+    accel = my_data[instruction_index][4]
+    hdg = my_data[instruction_index][5]
+
+    # use heading of most recent instruction without interpolation
+    new_heading = hdg
+    # calculate desired velocity by taking most recent time's velocity and interpolating with accel and time
+    new_velocity = vel + accel * (local_time - instruction_time)
+
+    drive_cmd = DriveCommand()
+    drive_cmd.heading = new_heading
+    drive_cmd.speed = new_velocity
+    command_pub.publish(drive_cmd)
+
+
+if __name__ == "__main__":
+    # seconds passed since epoch (global time, used to find local time at every call)
+    start_time = time.time()
+
+    # csv is generated with path, based on time passed since start
+    # will need to make sure to copy file into this directory after creating it in trajectory_gen
+    # will need to test how this works when the node is run from a launch file
+    instructions = genfromtxt('output_traj.csv', delimiter=',', skipheader=1, names="time,x,y,velocity,accel,heading")
+
+    # Initialize ROS node
+    rospy.init_node("nrc_drive_dr")
+
+    # Set up a publisher for publishing the drive command
+    command_pub = rospy.Publisher("/nrc/cmd", DriveCommand, generate_drive_command, queue_size=1)
```

Comment:
You'll need to remove `generate_drive_command` from this. It's being called by the timer. I also don't know why it let you include this here because I didn't think you could pass a function to a publisher, but I guess you can